src/libmongoc/tests/test-mongoc-client-side-encryption.c

```diff
@@ -3692,6 +3702,254 @@ explicit_encryption_destroy (ee_fixture *eef)
    bson_free (eef);
 }
 
+static void
+explicit_encryption_set_range_opts_int (
+   mongoc_client_encryption_encrypt_opts_t *eopts,
+   mongoc_client_encryption_encrypt_range_opts_t *rangeopts,
+   ee_fixture *eef)
+{
+   mongoc_client_encryption_encrypt_opts_set_keyid (eopts, &eef->key1ID);
+   mongoc_client_encryption_encrypt_opts_set_algorithm (
+      eopts, MONGOC_ENCRYPT_ALGORITHM_RANGEPREVIEW);
+
+   mongoc_client_encryption_encrypt_range_opts_set_sparsity (rangeopts, 1);
+   mongoc_client_encryption_encrypt_opts_set_contention_factor (eopts, 0);
+   bson_value_t min = {0};
+   min.value_type = BSON_TYPE_INT32;
+   min.value.v_int32 = 0;
+   bson_value_t max = {0};
+   max.value_type = BSON_TYPE_INT32;
+   max.value.v_int32 = 250;
+   mongoc_client_encryption_encrypt_range_opts_set_min_max (
+      rangeopts, min, max);
+   mongoc_client_encryption_encrypt_opts_set_range_opts (eopts, rangeopts);
+}
+
+static void
+test_explicit_encryption_range (void *unused)
+{
+   bson_error_t error;
+   bool ok;
+   mongoc_client_encryption_encrypt_opts_t *eopts;
+   mongoc_client_encryption_encrypt_range_opts_t *rangeopts;
+   bson_value_t plaintext = {0};
+   ee_fixture *eef = explicit_encryption_setup (true);
+
+   BSON_UNUSED (unused);
+
+   /* Use ``encryptedClient`` to insert 2 documents of the form ``{
+    * "encryptedInt": <insertPayload> }``. */
+   {
+      plaintext.value_type = BSON_TYPE_INT32;
+      plaintext.value.v_int32 = 6;
+      int i = 0;
+      for (i = 0; i < 2; i++) {
+         if (i > 0) {
+            plaintext.value.v_int32 = 30;
+         }
+         bson_value_t insertPayload;
+         bson_t to_insert = BSON_INITIALIZER;
+         eopts = mongoc_client_encryption_encrypt_opts_new ();
+         rangeopts = mongoc_client_encryption_encrypt_range_opts_new ();
+         explicit_encryption_set_range_opts_int (eopts, rangeopts, eef);
+
+         ok = mongoc_client_encryption_encrypt (
+            eef->clientEncryption, &plaintext, eopts, &insertPayload, &error);
+         ASSERT_OR_PRINT (ok, error);
+
+         ASSERT (
+            BSON_APPEND_VALUE (&to_insert, "encryptedInt", &insertPayload));
```

Comment:
```suggestion
            BSON_APPEND_VALUE (&to_insert, "encryptedInt", &insertPayload));
         ASSERT (BSON_APPEND_INT32 (&to_insert, "_id", i));
```

Suggest adding the `_id` field. This gives a field to sort when there are multiple results. This is related to a comment below.